test/inference/gibbs.jl

```diff
@@ -106,4 +106,90 @@ include(dir*"/test/test_utils/AllUtils.jl")
         alg = Gibbs(MH(:s), HMC(0.2, 4, :m))
         sample(model, alg, 100; callback = callback)
     end
+
+    @turing_testset "gibbs conditionals" begin
+        let α₀ = 2.0,
+            θ₀ = inv(3.0),
+            x = [1.5, 2.0]
+            
+            @model inverse_gdemo(x) = begin
+                λ ~ Gamma(α₀, θ₀)
+                m ~ Normal(0, √(1 / λ))
+                x .~ Normal(m, √(1 / λ))
+            end
+
+            function gdemo_statistics(x)
+                # The conditionals and posterior can be formulated in terms of the following statistics:
+                N = length(x) # number of samples
+                x̄ = mean(x) # sample mean
+                s² = var(x; mean=x̄, corrected=false) # sample variance
+                return N, x̄, s²
+            end
+
+            function cond_m(c)
+                N, x̄, s² = gdemo_statistics(x)
+                mₙ = N * x̄ / (N + 1)
+                λₙ = c.λ * (N + 1)
+                σₙ = √(1 / λₙ)
+                return Normal(mₙ, σₙ)
+            end
+
+            function cond_λ(c)
+                N, x̄, s² = gdemo_statistics(x)
+                αₙ = α₀ + (N - 1) / 2
+                βₙ = (s² * N / 2 + c.m^2 / 2 + inv(θ₀))
+                return Gamma(αₙ, inv(βₙ))
+            end
+
+            Random.seed!(100)
+            alg = Gibbs(
+                GibbsConditional(:m, cond_m),
+                GibbsConditional(:λ, cond_λ))
+            chain = sample(inverse_gdemo(x), alg, 3000)
+            check_numerical(chain, [:m, :λ], [7/6, 24/49], atol=0.1)
+        end
+
+        let π = [0.5, 0.5],
+            K = length(π),
+            λ = 5.0,
+            σ = 1.0,
+            x = [rand(10); 2 .+ rand(10)],
+            N = length(x)
+
+            @model mixture(x) = begin
+                μ ~ arraydist(Normal.(0, fill(λ, K)))
+                z ~ arraydist(Categorical.(fill(π, N)))
+                x ~ arraydist(Normal.(μ[z], σ))
+                
+                return x
+            end
+
+            # see http://www.cs.columbia.edu/~blei/fogm/2015F/notes/mixtures-and-gibbs.pdf
+            function cond_z(c)
+                μ = c.μ
+                function ϖ(ξ)
```

Comment:
Please use a slightly less unicode heavy notation here.

Reply:
Sure! Now why have I learn Greek then :(

Reply:
I would suggest to make a trade-off between readability and pretty notation. Using zeta is fine with me but more complex unicode letters (mu with hat and whatever) might be unnecessary.

Reply:
But are mu, sigma, and lambda OK?

Reply:
Of course. I would say, everything that is simply a greek letter should be fine. But I would restrain from using `μ̂` and so on.

Reply:
And I'm not sure it is necessary to use greek letters for function names. This was my initial reason for the comment here.

Reply:
> Using zeta is fine with me but more complex unicode letters (mu with hat and whatever) might be unnecessary.

That's xi.  Which is reason enough to change it :smile:

Reply:
Let's just agree that xi should never be used for anything, code or math.

Reply:
It's one of the nicest integration variables I know... way better than "anything prime" :P